src/main/java/com/google/cloud/spanner/pgadapter/utils/LazyInit.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.utils;
+
+import java.util.function.Supplier;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/** A simple not-thread-safe lazy initializer. */
+@NotThreadSafe
+public abstract class LazyInit<T> implements Supplier<T> {
```

Comment:
Do we use guava? If so, we can use `Suppliers.memoize` instead of creating this class

Reply:
Ah, yes. Thanks for the tip, I was looking around for a standard solution for this in the JDK libs and couldn't find anything, but I didn't think of looking into Guava. Changed.